frontend/awx/views/schedules/hooks/scheduleHelpers.tsx

```diff
@@ -68,3 +95,266 @@ export function useGetCreateRuleRoute() {
   }
   return createUrl;
 }
+export const promptonLaunchFieldNames: string[] = [
+  'ask_job_type_on_launch',
+  'ask_inventory_on_launch',
+  'ask_credential_on_launch',
+  'ask_execution_environment_on_launch',
+  'ask_instance_groups_on_launch',
+  'ask_scm_branch_on_launch',
+  'ask_forks_on_launch',
+  'ask_diff_mode_on_launch',
+  'ask_job_slice_count_on_launch',
+  'ask_labels_on_launch',
+  'ask_limit_on_launch',
+  'ask_skip_tags_on_launch',
+  'ask_tags_on_launch',
+  'ask_timeout_on_launch',
+  'ask_variables_on_launch',
+  'ask_verbosity_on_launch',
+];
+
+export function useGetPromptOnLaunchFields(
+  resourceForSchedule: JobTemplate | WorkflowJobTemplate | InventorySource | Project
+) {
+  const fields: ReactElement[] = [];
+  const promptFields: LaunchConfiguration = useWatch({
+    name: 'launchConfiguration',
+  }) as LaunchConfiguration;
+  const { t } = useTranslation();
+  const arrayedJobTags: { name: string; value: string; label: string }[] =
+    promptFields?.defaults?.job_tags === null
+      ? [{ name: '', value: '', label: '' }]
+      : promptFields?.defaults?.job_tags
+          ?.split(',')
+          .map((tag) => ({ name: tag, label: tag, value: tag }));
+  const arrayedSkipTags: { name: string; value: string; label: string }[] =
+    promptFields?.defaults?.skip_tags === null
+      ? [{ name: '', value: '', label: '' }]
+      : promptFields?.defaults?.skip_tags
+          ?.split(',')
+          .map((tag) => ({ name: tag, label: tag, value: tag }));
+  if (!resourceForSchedule || resourceForSchedule?.type === 'inventory_source') return [];
+  promptonLaunchFieldNames.forEach((fieldName: string) => {
+    if (promptFields && promptFields[fieldName] === true) {
```

Comment:
Consider reversing this if to un-nest the entire switch block.

```
if (!promptFields || promptFields[fieldName] !== true) {
  return;
}
switch (...) {
```

Reply:
Thats a good idea.  I've made that change